src/south/Modbus/Modbus.class.js

```diff
@@ -4,6 +4,21 @@ const net = require('net')
 const { getOptimizedScanModes } = require('./config/getOptimizedConfig')
 const ProtocolHandler = require('../ProtocolHandler.class')
 
+/**
+ * Read point value according to it's data type (UInt16, UInt32, etc)
+ * @param {Object} response Response of the modbus request
+ * @param {Object} point The point to read
+ * @param {Number} position Position of the point in the response
+ * @return {Number} Value stored at the specified address
+ */
+const readRegisterValue = (response, point, position) => {
+  if (response.body.constructor.name === 'ReadCoilsResponseBody' || response.body.constructor.name === 'ReadDiscreteInputsResponseBody') {
+    return response.body.valuesAsArray[position]
+  }
+  const funcName = `read${point.dataType}BE`
+  return response.body.valuesAsBuffer[funcName](position * 2)
```

Comment:
Not sure to understand the 'position*2' part, can you tell me more about it ?
A comment on that in the code may be useful

Reply:
By default, the jsmodbus library you use reads addresses values in Big Endian and unsigned int 16 bits. So in the response of the query, in the valuesAsArray field that you used to retrieve the value of the point, each index of the array corresponds to 16 bits. On the other hand in the valuesAsBuffer field, each index of the bufer corresponds to 1 byte or 8 bits. If we take the example where it is the value which is at the 5th index of the array of the valuesAsArray field, it will correspond to the index 10 of the valuesAsBuffer field, and instead of reading only one cell of the array you have to read 2, 4 or 8 according to the size of the value to retrieve (Int16, Int32, Int64). I will try to simplify my explanation and put it as a comment in the code, but is it clearer for you already?

Reply:
It's clearer, thanks. I missed that point in the jsmodbus library. Thank you for having fixed that !

My tests were kind of trivial, and following your answer, I think it must be improved to take this into account. I can improve it later, but if you can do it now, it would be perfect. !